src/DynamoCoreWpf/ViewModels/Core/StateMachine.cs

```diff
@@ -525,8 +525,12 @@ internal bool HandleLeftButtonDown(object sender, MouseButtonEventArgs e)
                             // a code block node being created, in which case we
                             // should keep the input focus on the code block to 
                             // avoid it being dismissed (with empty content).
-                            // 
-                            CreateCodeBlockNode(mouseDownPos);
+                            //
+                            // If Shift is pressed, CBN shouldn't be created.
+                            // Shift Modifier indicates, that user tries to call InCanvasSearch
+                            // by using Shift + DoubleClick.
+                            if (Keyboard.Modifiers != ModifierKeys.Shift)
+                                CreateCodeBlockNode(mouseDownPos);
```

Comment:
Do we need to exclude the `returnFocusToSearch = false;` below too? How about the rest of the method? Do we still need to proceed if SHIFT is held down?